### PR TITLE
Parser fix

### PR DESCRIPTION
**Pull request by PNieck**
## Description

This PR fixes `parser` library so that it accepts field value with `+` sign.

## Motivation and Context

More robust library.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `host-generic-pilot`, `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libs/parser/parser.c [hunk 1]
```diff
@@ -365,14 +366,12 @@ int parser_execute(parser_t *p, const char *path)
 		}
 	}
 
-	res = 0;
-
 	if (feof(file) == 0) {
-		fprintf(stderr, "%s: error on file parsing\n", FUNCTION_EXECUTE);
+		fprintf(stderr, "%s: error on parsing %s\n", FUNCTION_EXECUTE, path);
```
Comment (HBuczynski) [inline libs/parser/parser.c:370]: Please remove defines with function names, and use `__FUNCTION__`
```suggestion
		fprintf(stderr, "%s: error on parsing %s\n", __FUNCTION__, path);
```

**Review by HBuczynski — DISMISSED** [review]
LGTM

File: libs/parser/parser.h [hunk 2]
```diff
@@ -38,7 +42,7 @@ extern int parser_headerAdd(parser_t *p, const char *headerName, int (*converter
 
 
 /* Performs the parsing at file specified by `path` */
```
[the remark this answers is not in the document]
Comment (PNieck) [reply libs/parser/parser.h]: Done

**Comment by PNieck** [comment]
I have added additional tests for case when field doesn't belong to any header

**Review by niewim19 — APPROVED** [review]
LGTM

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**